### ISSUE TITLE
Can't Crack Old Wallet Spending Pin Android

### ISSUE THREAD

**Issue by moglimax** [gurnec/btcrecover#159]
Okay so my original backup of my wallet is dated 2016-07-10 although I think I made this back-up some time in February 2017 because at the time I just bought my first Bitcoin (I know form my transaction records on my exchange account). So I assume that date is when I downloaded the app, which I did do fairly early before purchasing my first Bitcoin. 

I bought a small amount of BTC and sent it from my exchange wallet to the wallet on my phone. When I tried to send it out of my android wallet it told me I needed to input a spending pin. I didn't even know I had set one. Then I made a backup of the wallet and wiped the app, only to forget my backup key. (all happened around February time)

Fortunately, yesterday I remembered that backup key. I updated the app and restored the wallet on my phone (synchronized from like 16 months). The spending key is still there so I went on Google and found btcrecover. I made another backup, with another password and ran that through btcrecover for the spending pin.

I ran btcrecover to search for 1-6 digits (%1,6d in the tokenlist file). Password possibilities where exhausted and I doubt I set a PIN of more than 6 digits. The whole process took about 2 hours 20 min. Now I'm halfway running a 7 digit recovery (10 hours spent) but nothing so far. However I read something about less key stretching depending on your app version or something... Also p=4096 (I don't know what that means) while others' have their p=65536. 

**Now here's the real kicker that has me confused.** Because of this key space thing, thought 'okay, since I have the original backup, along with the password, I'll run that one through btcrecover.'
 But the problem is, when asked for the password to the original backup, I enter it and get back the message 'can't decrypt wallet (wrong password?)'. Now I know I entered the right password because it worked in recovering to my phone and successfully recovered it onto my phone twice with the same password and I wrote it down so I would not forget. I've done this process multiple times in btcrecover carefully to ensure typos aren't the problem. 

So in short, I can run the new backup (same wallet and spending pin as the original, but different backup password) through btcrecover when inputting the password for decryption, however I cannot run my original backup file through btcrecover even though I input the correct password. 

Your help would be really appreciated. Thanks in advance.

**Comment by amkida4real** [comment]
I guess by now you are through with the 7 digit recovery and I hope you have found your pin. 
I just created a new issue #169 . Please check if you can assist in anyway because you seem to have gotten it right. Thanks!

**Comment by moglimax** [comment]
Unfortunately, I haven't yet. Cleared all 7 digit combinations and no pin. It might be 8 because on the app, once you type 8 digits the password is described as 'strong'. So I might have been taunted into making 8 digits. But the problem I'm stuck with is that it counts the combinations, then freezes at 22 300 000. The total amount of combinations it should reach is 100 000 000 000 combinations.

**Comment by elkiwi** [comment]
@moglimax you need to run the script with:

--no-eta --no-dupchecks 

then it won't get killed for whatever resource ran out counting.

**Comment by nori19870522** [comment]
I am Japanese and do password analysis.
https://keychainx.io/ 

This is Robert's article.
https://news.bitstarz.com/bitstarz-news-interview-robert-rhodin-keychainx-ceo  

I may be able to find the password you lost. Do you need help with your login?

**Comment by moglimax** [comment]
@elkiwi hey man. thanks for the tip. I totally forgot about this thread and I actually never tried your suggestion yet. I'm going to try it and I'll get back to you. Thanks.

**Comment by btcexchanging** [comment]
if you haven't recovered your wallet, i can help you recover your wallet.
I have recovered most of the common wallets and recovery phrases for many users
you can reach me here at my telegram account:  @walletrecover

**Comment by viladmir2022** [comment]
hi, if you still have wallet problem. I can recover your lost wallet.
I have powerful servers and excellent software that can find any password.
and my fee is only 15-25% of the recovered fund
here is my telegram id: @viladmir2022  and Gmail: walletsrecoveryservice@gmail.com
